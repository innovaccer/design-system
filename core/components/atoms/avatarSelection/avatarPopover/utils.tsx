import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  withSearch?: boolean,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  const listItems = listRef?.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else if (
    (withSearch && index === 0 && direction === 'up') ||
    (withSearch && index === listItems.length - 1 && direction === 'down')
  ) {
    const searchInput = listRef.current.querySelector('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput.focus();
    setFocusedOption && setFocusedOption(searchInput);
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;

    const targetOption = listItems[index];

    (targetOption as HTMLElement).focus();
    setFocusedOption && setFocusedOption(targetOption);
    targetOption.scrollIntoView({ block: 'center' });
  }
};

export const handleInputKeyDown = (
  event: React.KeyboardEvent,
  listRef: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      targetOption = listItems[listItems.length - 1];
      break;
    case 'ArrowDown':
      event.preventDefault();
      targetOption = listItems[0];
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }

  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
