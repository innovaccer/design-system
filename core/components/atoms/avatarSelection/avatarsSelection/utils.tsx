import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'Enter':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  if (withSearch && position === 'down') {
    const searchInput = listRef.current?.querySelectorAll('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput && searchInput[0]?.focus();
    setFocusedOption?.(searchInput);
    return;
  }

  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
