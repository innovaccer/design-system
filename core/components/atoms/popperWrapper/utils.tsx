// Helper function to convert CSS class names to style objects
export const createStyleFromClass = (className: string) => {
  if (!className) return {};

  // Create a temporary element to get computed styles
  const tempElement = document.createElement('div');
  tempElement.className = className;
  document.body.appendChild(tempElement);

  const computedStyle = window.getComputedStyle(tempElement);
  const style: React.CSSProperties = {};

  // Extract relevant animation properties
  const animation = computedStyle.animation;
  const animationDuration = computedStyle.animationDuration;
  const animationTimingFunction = computedStyle.animationTimingFunction;
  const animationFillMode = computedStyle.animationFillMode;
  const opacity = computedStyle.opacity;
  const transform = computedStyle.transform;

  if (animation && animation !== 'none') {
    style.animation = animation;
  }
  if (animationDuration && animationDuration !== '0s') {
    style.animationDuration = animationDuration;
  }
  if (animationTimingFunction && animationTimingFunction !== 'ease') {
    style.animationTimingFunction = animationTimingFunction;
  }
  if (animationFillMode && animationFillMode !== 'none') {
    style.animationFillMode = animationFillMode;
  }
  if (opacity && opacity !== '1') {
    style.opacity = opacity;
  }
  if (transform && transform !== 'none') {
    style.transform = transform;
  }

  document.body.removeChild(tempElement);
  return style;
};
