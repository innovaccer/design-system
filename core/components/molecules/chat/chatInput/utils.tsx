import { MessageType } from './ChatInput';

export const positionMentionPopup = (
  textareaRef: React.RefObject<HTMLDivElement>,
  setMentionPosition: React.Dispatch<React.SetStateAction<{ top: number; left: number }>>
) => {
  if (!textareaRef.current) return;

  const selection = window.getSelection();
  if (!selection || !selection.rangeCount) return;

  const range = selection.getRangeAt(0);
  const rect = range.getBoundingClientRect();
  const editorRect = textareaRef.current.getBoundingClientRect();

  // Calculate position based on the cursor position
  let top = rect.bottom + window.scrollY;
  let left = rect.left + window.scrollX;

  // Adjust position if it would go off-screen
  if (top + 150 > window.innerHeight) {
    top = rect.top + window.scrollY - 150;
  }
  if (left + 200 > window.innerWidth) {
    left = window.innerWidth - 200;
  }

  // Adjust position relative to the editor
  top -= editorRect.top + window.scrollY - 12;
  left -= editorRect.left + window.scrollX - 8;

  setMentionPosition({ top, left });
};

export const extractMessageData = (textareaRef: React.RefObject<HTMLDivElement>): MessageType[] => {
  const elements = textareaRef.current?.childNodes;

  return Array.from(elements || []).map((item: any) => {
    if (item?.dataset?.type === 'mention') {
      const mentionValue = JSON.parse(item.dataset.content);
      return { type: 'mention', content: mentionValue?.data, nodeName: item.nodeName };
    }

    // handles new line by pressing enter
    else if (item.innerText === '\n' && item.nodeName === 'DIV') {
      return { type: 'newline', content: '\n', nodeName: item.nodeName };
    }

    // handles line break by pressing shift + enter
    else if (item.nodeName === 'BR') {
      return { type: 'linebreak', content: '<br>', nodeName: item.nodeName };
    }

    return { type: 'text', content: item.innerText || item.textContent, nodeName: item.nodeName };
  });
};
