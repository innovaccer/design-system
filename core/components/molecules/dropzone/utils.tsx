const isIe = (userAgent: string) => {
  return userAgent.indexOf('MSIE') !== -1 || userAgent.indexOf('Trident/') !== -1;
};

const isEdge = (userAgent: string) => {
  return userAgent.indexOf('Edge/') !== -1;
};

export const accepts = (file: File, acceptedFiles?: string | string[]) => {
  if (file && acceptedFiles) {
    const acceptedFilesArray = Array.isArray(acceptedFiles) ? acceptedFiles : acceptedFiles.split(',');
    const fileName = file.name || '';
    const mimeType = (file.type || '').toLowerCase();
    const baseMimeType = mimeType.replace(/\/.*$/, '');

    if (!mimeType) {
      return true;
    }

    return acceptedFilesArray.some((type) => {
      const validType = type.trim().toLowerCase();
      if (validType.charAt(0) === '.') {
        return fileName.toLowerCase().endsWith(validType);
      }
      if (validType.endsWith('/*')) {
        return baseMimeType === validType.replace(/\/.*$/, '');
      }
      return mimeType === validType;
    });
  }
  return true;
};

export const isPropagationStopped = (event: any) => {
  if (typeof event.isPropagationStopped === 'function') {
    return event.isPropagationStopped();
  }
  if (typeof event.cancelBubble !== 'undefined') {
    return event.cancelBubble;
  }
  return false;
};

export const isEvtWithFiles = (event: any) => {
  if (!event.dataTransfer) {
    return !!event.target && !!event.target.files;
  }

  return Array.prototype.some.call(
    event.dataTransfer.types,
    (type) => type === 'Files' || type === 'application/x-moz-file'
  );
};

export const isKindFile = (item: null | Record<string, unknown>) => {
  return typeof item === 'object' && item !== null && item.kind === 'file';
};

export const onDocumentDragOver = (event: DragEvent) => {
  event.preventDefault();
};

export const isIeOrEdge = (userAgent = window.navigator.userAgent) => {
  return isIe(userAgent) || isEdge(userAgent);
};

type functype = (event: Event, ...args: unknown[]) => void;

export const composeEventHandlers = (...fns: functype[]) => {
  return (event: Event, ...args: unknown[]) =>
    fns.some((fn) => {
      if (!isPropagationStopped(event) && fn) {
        fn(event, ...args);
      }
      return isPropagationStopped(event);
    });
};

export const reducer = (state: any, action: any) => {
  switch (action.type) {
    case 'focus':
      return {
        ...state,
        isFocused: true,
      };
    case 'blur':
      return {
        ...state,
        isFocused: false,
      };
    case 'openDialog':
      return {
        ...state,
        isFileDialogActive: true,
      };
    case 'closeDialog':
      return {
        ...state,
        isFileDialogActive: false,
      };
    case 'setDraggedFiles':
      /* eslint no-case-declarations: 0 */
      const { isDragActive, draggedFiles } = action;
      return {
        ...state,
        draggedFiles,
        isDragActive,
      };
    case 'setFiles':
      return {
        ...state,
        acceptedFiles: action.acceptedFiles,
        fileRejections: action.fileRejections,
      };
    case 'reset':
      return {
        ...state,
        isFileDialogActive: false,
        isDragActive: false,
        draggedFiles: [],
        acceptedFiles: [],
        fileRejections: [],
      };
    default:
      return state;
  }
};
