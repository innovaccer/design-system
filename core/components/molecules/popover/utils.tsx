import { Placement } from '@floating-ui/react';
import { PositionType as Position } from '@/common.type';

export const mapPositionToPlacement = (position: Position): Placement => {
  const positionMap: Record<Position, Placement> = {
    auto: 'bottom',
    'auto-start': 'bottom-start',
    'auto-end': 'bottom-end',
    'top-start': 'top-start',
    top: 'top',
    'top-end': 'top-end',
    'right-start': 'right-start',
    right: 'right',
    'right-end': 'right-end',
    'bottom-end': 'bottom-end',
    bottom: 'bottom',
    'bottom-start': 'bottom-start',
    'left-end': 'left-end',
    left: 'left',
    'left-start': 'left-start',
  };

  return positionMap[position] || 'bottom';
};
