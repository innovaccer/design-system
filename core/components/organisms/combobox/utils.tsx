import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  inputTriggerRef?: any,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  multiSelect?: boolean,
  listRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef);
      break;
    case 'Enter':
      handleEnterKey(focusedOption, multiSelect, inputTriggerRef, listRef, setFocusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      inputTriggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (
  focusedOption: Element | undefined,
  multiSelect?: boolean,
  inputTriggerRef?: any,
  listRef?: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>
) => {
  (focusedOption as HTMLElement)?.click();
  if (!multiSelect) {
    inputTriggerRef.current.focus();
  } else {
    // to focus first option by default when last option is selected
    const listItems = listRef.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
    const listArr = Array.from(listItems);
    const index = listArr.findIndex((item) => {
      return item == focusedOption;
    });

    if (index === listArr.length - 1) {
      (listItems[0] as HTMLElement).focus();
      setFocusedOption && setFocusedOption(listItems[0]);
      listItems[0].scrollIntoView({ block: 'center' });
    }
  }
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any
) => {
  const listItems = listRef.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;
  }

  const targetOption = listItems[index];

  (targetOption as HTMLElement).focus();
  setFocusedOption && setFocusedOption(targetOption);
  targetOption?.scrollIntoView?.({ block: 'center' });
};
