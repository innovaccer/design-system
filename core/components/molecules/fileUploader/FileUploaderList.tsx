import * as React from 'react';
import classNames from 'classnames';
import FileUploaderItem, { FileItem } from './FileUploaderItem';
import { BaseProps, extractBaseProps } from '@/utils/types';

export interface FileUploaderListProps extends BaseProps {
  /**
   * <pre style="font-family: monospace; font-size: 13px; background: #f8f8f8">
   * Array of FileItem object.
   * FileItem:
   * {
   *    file: File;
   *    id?: any;
   *    status: FileStatus;
   *    progress: number;
   *    errorMessage: string;
   * }
   *
   * | Name | Description | Default |
   * | --- | --- | --- |
   * | file | Attached file | |
   * | id | Unique id/index of file | |
   * | status | Status of file | 'completed' |
   * | progress | Progress percent of `Progress Ring` | 0 |
   * | errorMessage | Error Message to be shown when status is `error` | 'Network Error' |
   * </pre>
   */
  fileList: FileItem[];
  /**
   * Callback called when file item is clicked
   */
  onClick?: (file: File, id?: any) => void;
  /**
   * Callback called when remove icon is clicked
   */
  onDelete?: (file: File, id?: any) => void;
  /**
   * Callback called when retry icon is clicked
   */
  onRetry?: (file: File, id?: any) => void;
}

export const FileUploaderList = (props: FileUploaderListProps) => {
  const { fileList, onClick, onDelete, onRetry, className } = props;

  const baseProps = extractBaseProps(props);

  const FileListClass = classNames(
    {
      ['FileUploaderList']: true,
    },
    className
  );

  if (fileList.length === 0) return null;

  return (
    <div {...baseProps} className={FileListClass} data-test="DesignSystem-FileUploader--List">
      {fileList.map((fileName, i) => (
        <FileUploaderItem key={i} onDelete={onDelete} onRetry={onRetry} onClick={onClick} {...fileName} />
      ))}
    </div>
  );
};

FileUploaderList.defaultProps = {
  fileList: [],
};

FileUploaderList.displayName = 'FileUploaderList';

export default FileUploaderList;
