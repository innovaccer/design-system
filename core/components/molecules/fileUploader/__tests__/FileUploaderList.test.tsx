import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';
import FileUploaderList, { FileUploaderListProps as Props } from '../FileUploaderList';

const fileList: Props['fileList'] = [
  {
    file: (File = {
      name: 'Audio File.mp3',
      size: '3 MB',
      type: 'audio',
    } as any),
    status: 'uploading',
    progress: 45,
    id: 1,
  },
  {
    file: (File = {
      name: 'Video File.mp4',
      size: '3 MB',
      type: 'video',
    } as any),
    status: 'completed',
    id: 2,
  },
  {
    file: (File = {
      name: 'Image File.jpeg',
      size: '3 MB',
      type: 'image',
    } as any),
    status: 'error',
    errorMessage: 'Network failure',
    id: 3,
  }
];

const FunctionValue = jest.fn();

describe('FileUploaderList component', () => {
  const mapper: Record<string, any> = {
    fileList: valueHelper(fileList, { required: true }),
    onClick: valueHelper(FunctionValue, { required: true }),
    onDelete: valueHelper(FunctionValue, { required: true }),
    onRetry: valueHelper(FunctionValue, { required: true })
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { asFragment } = render(
        <FileUploaderList
          {...attr}
        />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('FileUploaderList component prop:fileList', () => {
  it('renders component if fileList is empty', () => {
    const { queryByTestId } = render(
      <FileUploaderList />
    );
    expect(queryByTestId(/DesignSystem-FileUploader--List/i)).toBeNull();
  });

  it('renders component if fileList is not empty', () => {
    const { getByTestId, getAllByTestId } = render(
      <FileUploaderList
        fileList={fileList}
        onDelete={FunctionValue}
        onRetry={FunctionValue}
        onClick={FunctionValue}
      />
    );
    expect(getByTestId('DesignSystem-FileUploader--List')).toBeInTheDocument();
    expect(getAllByTestId('DesignSystem-FileUploader--Item')).toHaveLength(fileList.length);
  });
});

describe('FileUploaderList component Event Handler', () => {
  it('check for onClick Event Handler', () => {
    const { getAllByTestId } = render(
      <FileUploaderList
        fileList={fileList}
        onClick={FunctionValue}
      />
    );
    fireEvent.click(getAllByTestId('DesignSystem-FileUploader--Item')[0]);
    expect(FunctionValue).toHaveBeenCalled();
  });

  it('check for onDelete Event Handler', () => {
    const { getAllByTestId } = render(
      <FileUploaderList
        fileList={fileList}
        onDelete={FunctionValue}
      />
    );
    fireEvent.click(getAllByTestId('DesignSystem-FileUploader--CancelButton')[0]);
    expect(FunctionValue).toHaveBeenCalled();
  });

  it('check for onRetry Event Handler', () => {
    const { getByTestId } = render(
      <FileUploaderList
        fileList={[fileList[2]]}
        onRetry={FunctionValue}
      />
    );
    fireEvent.click(getByTestId('DesignSystem-Button'));
    expect(FunctionValue).toHaveBeenCalled();
  });

  it('check for Error Caption', () => {
    const { getByTestId } = render(
      <FileUploaderList
        fileList={[fileList[2]]}
        onRetry={FunctionValue}
      />
    );
    expect(getByTestId('DesignSystem-Caption')).toBeInTheDocument();
    expect(getByTestId('DesignSystem-Caption').textContent).toMatch('Network failure');
  });
});
