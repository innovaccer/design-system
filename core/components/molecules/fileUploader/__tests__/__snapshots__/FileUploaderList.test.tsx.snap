// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploaderList component 
  fileList: [{"file":{"name":"Audio File.mp3","size":"3 MB","type":"audio"},"status":"uploading","progress":45,"id":1},{"file":{"name":"Video File.mp4","size":"3 MB","type":"video"},"status":"completed","id":2},{"file":{"name":"Image File.jpeg","size":"3 MB","type":"image"},"status":"error","errorMessage":"Network failure","id":3}], onClick: "[Function]", onDelete: "[Function]", onRetry: "[Function]"
 1`] = `
<DocumentFragment>
  <div
    class="FileUploaderList"
    data-test="DesignSystem-FileUploader--List"
  >
    <div
      class="FileUploaderItem"
      data-test="DesignSystem-FileUploader--Item"
    >
      <div
        class="FileUploaderItem-file"
      >
        <span
          class="Text Text--subtle Text--regular FileUploaderItem-text"
          data-test="DesignSystem-Text"
        >
          Audio File.mp3
        </span>
        <div
          class="d-flex align-items-center"
        >
          <svg
            class="Ring Ring--small mr-4"
            data-test="DesignSystem-ProgressRing"
            viewBox="0 0 50 50"
          >
            <circle
              class="Ring-background"
              cx="25"
              cy="25"
              fill="none"
              r="20"
              stroke-dasharray="125.66370614359172 125.66370614359172"
              stroke-width="8"
            />
            <circle
              class="Ring-indicator"
              cx="25"
              cy="25"
              data-test="DesignSystem-ProgressRing--Circle"
              fill="none"
              r="20"
              stroke-dasharray="125.66370614359172 125.66370614359172"
              stroke-dashoffset="69.11503837897544"
              stroke-width="8"
            />
          </svg>
          <button
            class="Button Button--regular Button--regularSquare Button--transparent"
            data-test="DesignSystem-FileUploader--CancelButton"
            tabindex="0"
          >
            <div
              class="Button-icon"
            >
              <i
                class="material-icons material-icons-round Icon Icon--default"
                data-test="DesignSystem-Button--Icon"
                style="font-size: 16px; width: 16px;"
              >
                close_round
              </i>
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="FileUploaderItem"
      data-test="DesignSystem-FileUploader--Item"
    >
      <div
        class="FileUploaderItem-file"
      >
        <span
          class="Text Text--default Text--regular FileUploaderItem-text"
          data-test="DesignSystem-Text"
        >
          Video File.mp4
        </span>
        <div
          class="d-flex align-items-center"
        >
          <button
            class="Button Button--regular Button--regularSquare Button--transparent"
            data-test="DesignSystem-FileUploader--CancelButton"
            tabindex="0"
          >
            <div
              class="Button-icon"
            >
              <i
                class="material-icons material-icons-round Icon Icon--default"
                data-test="DesignSystem-Button--Icon"
                style="font-size: 16px; width: 16px;"
              >
                close_round
              </i>
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="FileUploaderItem"
      data-test="DesignSystem-FileUploader--Item"
    >
      <div
        class="FileUploaderItem-file"
      >
        <span
          class="Text Text--subtle Text--regular FileUploaderItem-text"
          data-test="DesignSystem-Text"
        >
          Image File.jpeg
        </span>
        <div
          class="d-flex align-items-center"
        >
          <button
            class="Button Button--regular Button--regularSquare Button--transparent mr-2"
            data-test="DesignSystem-Button"
            tabindex="0"
          >
            <div
              class="Button-icon"
            >
              <i
                class="material-icons material-icons-round Icon Icon--default"
                data-test="DesignSystem-Button--Icon"
                style="font-size: 16px; width: 16px;"
              >
                refresh_round
              </i>
            </div>
          </button>
          <button
            class="Button Button--regular Button--regularSquare Button--transparent"
            data-test="DesignSystem-FileUploader--CancelButton"
            tabindex="0"
          >
            <div
              class="Button-icon"
            >
              <i
                class="material-icons material-icons-round Icon Icon--default"
                data-test="DesignSystem-Button--Icon"
                style="font-size: 16px; width: 16px;"
              >
                close_round
              </i>
            </div>
          </button>
        </div>
      </div>
      <div
        class="Caption"
        data-test="DesignSystem-Caption"
      >
        <div
          class="Caption-icon"
        >
          <i
            class="material-icons material-icons-round Icon Icon--alert"
            style="font-size: 14px; width: 14px;"
          >
            error_round
          </i>
        </div>
        <span
          class="Text Text--destructive Text--medium Text--small"
          data-test="DesignSystem-Text"
        >
          Network failure
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
